internal/services/storage/storage_account_resource.go

```diff
@@ -2146,10 +2207,28 @@ func resourceStorageAccountRead(d *pluginsdk.ResourceData, meta interface{}) err
 
 	if supportLevel.supportBlob {
 		blobClient := storageClient.BlobServicesClient
-		blobProps, err := blobClient.GetServiceProperties(ctx, id.ResourceGroupName, id.StorageAccountName)
-		if err != nil {
-			return fmt.Errorf("reading blob properties for %s: %+v", *id, err)
+		var blobProps storage.BlobServiceProperties
+
+		// wait for blob service endpoint to become available
+		log.Printf("[DEBUG] waiting for %s blob service to become available", *id)
+		stateConf := &pluginsdk.StateChangeConf{
+			Pending: []string{"Pending"},
+			Target:  []string{"Available"},
+			Refresh: func() (interface{}, string, error) {
+				blobProps, err = blobClient.GetServiceProperties(ctx, id.ResourceGroupName, id.StorageAccountName)
+				if err != nil {
+					return handleStorageServiceError("reading", "blob", err)
+				}
+				return blobProps, "Available", nil
+			},
+			MinTimeout: 10 * time.Second,
+			Timeout:    d.Timeout(pluginsdk.TimeoutRead),
 		}
+
+		if _, err = stateConf.WaitForStateContext(ctx); err != nil {
+			return fmt.Errorf("reading %s blob service properties: %+v", *id, err)
+		}
```

Comment:
the Create function shouldn't be returning until the Storage Account has finished propagating - as such rather than polling on this we can assume this exists at this point (else we'd still be creating the Storage Account) - given this can we remove this poller?

Reply:
Hi @tombuildsstuff, at first i also been convinced this will work. Unfortunately here is example which apply to all service endpoints:

1. Create function will reach the service endpoint only if `blob_properties` block is defined. So this pooling may be just skipped.

```golang
if val, ok := d.GetOk("blob_properties"); ok {
...
```

2. Create function return Read function which **always** try to get service properties from the endpoint

```
if supportLevel.supportBlob {
...
```

And if `blob_properties` is not defined it will cause the 404 error because of no polling.

So i left the poling function in Read function and that seems to be working.